server/pkg/schema/field_asset.go

```diff
@@ -10,14 +10,6 @@ type FieldAsset struct {
 	defaultValue *id.AssetID
 }
 
-// NewFieldAsset
-// TODO: check if its ok to remove this
-func NewFieldAsset() *FieldAsset {
-	return &FieldAsset{
-		defaultValue: nil,
-	}
-}
-
 func FieldAssetFrom(id *id.AssetID) *FieldAsset {
 	return &FieldAsset{
 		defaultValue: id,
```

Comment:
Clone ID (`id.Clone()`) because it allows modifying ID after initialization

```go
id := id.NewAssetID().Ref()
tp := FieldAssetFrom(id)
*id = id.NewAssetID().Ref() // changed!
```